Source/ARTPush.m

```diff
@@ -235,6 +239,16 @@ - (void)deactivate {
     }];
 }
 
+- (void)getState:(nonnull ARTPushStateCallback)callback {
+    [_admin.deviceRegistrations get:_rest.device.id callback:^(ARTDeviceDetails *device, ARTErrorInfo *error) {
```

Comment:
Well, it depends on what we decide `getState` should be: method for admins or simple users. AFAIK we didn't yet, neither server team did (although I don't understand why admins would want this method, because it's just a wrapper for another one).

Reply:
When you said:
> or it's a cocoa client implementation mistake to put this method into Admin's namespace.

It looked to me that you were referencing the **existing** implementation, not `getState` (which actually has been renamed to `getDevicePushDetails`)

We have discussed that this method should not need push admin, and realtime server changes should be made. In the mean time though, can the implementation of the code be improved? It appears like the `_admin` instance is being used, which would be very confusing. Perhaps this needs to be refactored?

Reply:
> It appears like the _admin instance is being used, which would be very confusing

As we can see here in java https://github.com/ably/ably-java/blob/main/lib/src/main/java/io/ably/lib/push/PushBase.java
`DeviceRegistrations` is on the same level as `Admin`, and the latter contains nothing except `publish` method. So I assume `ARTPushAdmin` in cocoa contains `ARTPushDeviceRegistrations` and `ARTPushChannelSubscriptions` by mistake. So I moved it from there into the `ARTPush` namespace - https://github.com/ably/ably-cocoa/pull/1186/commits/b6f1eaf641b37019fa0d7d81192e661f3f463e87
@QuintinWillison @lawrence-forooghian  @lukasz-szyszkowski

Reply:
Moved files out of `Admin` group as well https://github.com/ably/ably-cocoa/pull/1186/commits/d447d2c27eff787806f669971cc430b107134893

Reply:
@QuintinWillison if you are not against these changes please resolve this conversation since Ben started it.

Reply:
@lawrence-forooghian could you please take a look and resolve this conversation if you feel it can be. I've not got enough context or time to get involved right now.

Reply:
@maratal is your question "can we move `deviceRegistrations` and `channelSubscriptions` from `ARTPushAdmin` to `ARTPush`"? If so, I think [this comment](https://github.com/ably/ably-cocoa/pull/1186#discussion_r786795482) of mine addresses it.